ToolBox/Program.cs

```diff
@@ -302,5 +307,71 @@ private static string GetParameterOrDefault(IReadOnlyDictionary<string, object>
 
             return value.ToString();
         }
+
+        /// <summary>
+        /// Loads values from JSON text file
+        /// </summary>
+        private static Dictionary<string, object> LoadOptionsObjectFromConfigurationFile(string path)
+        {
+            var optionsObject = new Dictionary<string, object>();
+            try
+            {
+                // Read the text from json and fill that in a specifal object that
+                // will hold the values corresponding to json file contents
+                var toolBoxOptionsConfig = JsonConvert.DeserializeObject<ToolBoxOptionsConfiguration>(File.ReadAllText(path));
```

Comment:
Why not use `JObject` here? If the goal is to simply get a `Dictionary<string, object>` then the `ToolBoxOptionsConfiguration` class isn't needed. I think the following reduces this entire method to a couple lines:

``` csharp
var contents = File.ReadAllText(path);
var jObject = JObject.Parse(contents);
var optionsObject = jObject.ToObject<Dictionary<string, object>>();
```

Reply:
I tried your suggestion - thanks ! - it appears that no matter what values are specified in config file Json.NET will deserialize it into a Dictionary where all the properties from file will be contained as distionary key elements. At the same time, the program expects that dictionary contains keys-values only for those properties that have representative information - value is neither null nor empty. Supporting the own parser does let to achieve this. Consider this line in Program.cs:
https://github.com/QuantConnect/Lean/blob/master/ToolBox/Program.cs#L72
```
var toDate = optionsObject.ContainsKey("to-date")
                    ? DateTime.ParseExact(optionsObject["to-date"].ToString(), "yyyyMMdd-HH:mm:ss", CultureInfo.InvariantCulture)
                    : DateTime.UtcNow;
```
If we used the regular deserializer our dictionary would contain "to-date" and this line would break down the exectuion.

Reply:
Great points 👍